core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -69,9 +70,11 @@ protected ResourcePool(int maxCapacity, ConcurrentLinkedQueue<T> resources) {
    * the pool.
    *
    * @return a resource taken from the pool
+   * @throws IOException if an interruption occurs while acquiring resources
    */
-  public T acquire() {
-    return acquire(null, null);
+  @Override
+  public T acquire() throws IOException {
+    return acquire(WAIT_INDEFINITELY, null);
```

Comment:
So no branch, and ensuring the input must be positive? I think it's better to file a new jira to refactor the implementation.